src/core/ack_tracker.c

```diff
@@ -96,6 +96,58 @@ QuicAckTrackerAddPacketNumber(
         !RangeUpdated;
 }
 
+//
+// This function implements the logic defined in Section 6.2 [draft-ietf-quic-frequence-10]
+// to determine if the reordering threshold has been hit.
+//
+BOOLEAN
+QuicAckTrackerDidHitReorderingThreshold(
+    _In_ QUIC_ACK_TRACKER* Tracker,
+    _In_ uint8_t ReorderingThreshold
+    )
+{
+    if (ReorderingThreshold == 0) {
+        return FALSE;
+    }
+
+    const uint64_t LargestUnacked = QuicRangeGetMax(&Tracker->PacketNumbersToAck);
+    uint64_t LargestReported = 0; // The largest packet number that could be declared lost 
+
+    if (Tracker->LargestPacketNumberAcknowledged >= ReorderingThreshold &&
+        Tracker->LargestPacketNumberAcknowledged - ReorderingThreshold + 1 <= LargestUnacked) {
```

Comment:
If this line (117) is false, does it make sense to do anything? It means we haven't received enough packets between largest ACK and largest Unacked to trigger the threshold, right? Why is this simply not a sub-if block in this to return false?

Reply:
Yes you are right